### ISSUE TITLE
[Bug]: BadRequestError(400, 'search_phase_execution_exception', 'failed to create query: [knn] queries are only supported on [dense_vector] fields')

### ISSUE THREAD

**Issue by rhondasu2002** [run-llama/llama_index#11417]
### Bug Description

After I installed elasticsearch (version 8.9.0) using docker-ce on Ubuntu22.04, I tried to call the following code to store the index on elasticsearch.
```python
import logging
import sys
import os
import openai
from llama_index.vector_stores.elasticsearch import ElasticsearchStore
from llama_index.core import StorageContext, ServiceContext
from llama_index.core import VectorStoreIndex, SimpleDirectoryReader
from llama_index.embeddings.openai import OpenAIEmbedding
from llama_index.core import Settings
from llama_index.core.node_parser.text import SentenceSplitter
from langchain.chat_models import ChatOpenAI
logging.basicConfig(stream=sys.stdout, level=logging.INFO)
logging.getLogger().addHandler(logging.StreamHandler(stream=sys.stdout))
import getpass
import openai


os.environ["OPENAI_API_KEY"] = getpass.getpass("OpenAI API Key:")

openai.api_key = os.environ["OPENAI_API_KEY"]



es = ElasticsearchStore(
    index_name="my_index",
    es_url="http://localhost:9200",
)

documents = SimpleDirectoryReader("./data/paul_graham/").load_data()


vector_store = ElasticsearchStore(
    es_url="http://localhost:9200",
    index_name="paul_graham",
)
storage_context = StorageContext.from_defaults(vector_store=vector_store)
index = VectorStoreIndex.from_documents(
    documents, storage_context=storage_context
)

query_engine = index.as_query_engine()
response = query_engine.query("what were his investments in Y Combinator?")
print(response)
```
But I encountered this error:
```shell
BadRequestError(400, 'search_phase_execution_exception', 'failed to create query: [knn] queries are only supported on [dense_vector] fields')
```
So I checked my elasticsearch database and found that the type of "vector_field" is "float", but it should actually be "dense_vector".
So I checked the file llama_index/llama-index-integrations/vector_stores/llama-index-vector-stores-elasticsearch/llama_index/vector_stores/elasticsearch/base.py. In line 242 of the file, that is, the "_create_index_if_not_exists" function:
```
if self.client.indices.exists(index=index_name):
```
This will always return true. In fact, in the _init_ function(line176), the index has been created, but no data has been stored in it. So self.client.indices.exists mistakenly regards the empty index as having been successfully created (that is, created and stored data in it). Therefore, the else branch after line 246 will never be executed. So the data will never be created and stored in the index. When I try to execute the else branch after 246, my elasticsearch index is created normally and can be queried without the previous error.

### Version

0.10.12

### Steps to Reproduce

1. Install docker-ce on ubuntu22.04
2. Install elasticsearch 8.9.0
3. Create and start the container
```shell
docker run -p 9200:9200 \  -e "discovery.type=single-node" \  -e "xpack.security.enabled=false" \  -e "xpack.security.http.ssl.enabled=false" \  -e "xpack.license.self_generated.type=trial" \  docker.elastic.co/elasticsearch/elasticsearch:8.9.0
```
4. Install llama-index and llama-index-vector-stores-elasticsearch
```shell
pip install llama-index-vector-stores-elasticsearch
pip install llama-index
```
5. Call the following code
```python
import logging
import sys
import os
import openai
from llama_index.vector_stores.elasticsearch import ElasticsearchStore
from llama_index.core import StorageContext, ServiceContext
from llama_index.core import VectorStoreIndex, SimpleDirectoryReader
from llama_index.embeddings.openai import OpenAIEmbedding
from llama_index.core import Settings
from llama_index.core.node_parser.text import SentenceSplitter
from langchain.chat_models import ChatOpenAI
logging.basicConfig(stream=sys.stdout, level=logging.INFO)
logging.getLogger().addHandler(logging.StreamHandler(stream=sys.stdout))
import getpass
import openai


os.environ["OPENAI_API_KEY"] = getpass.getpass("OpenAI API Key:")

openai.api_key = os.environ["OPENAI_API_KEY"]



es = ElasticsearchStore(
    index_name="my_index",
    es_url="http://localhost:9200",
)

documents = SimpleDirectoryReader("./data/paul_graham/").load_data()


vector_store = ElasticsearchStore(
    es_url="http://localhost:9200",
    index_name="paul_graham",
)
storage_context = StorageContext.from_defaults(vector_store=vector_store)
index = VectorStoreIndex.from_documents(
    documents, storage_context=storage_context
)

query_engine = index.as_query_engine()
response = query_engine.query("what were his investments in Y Combinator?")
print(response)
```

**Comment by nerdai** [comment]
Thanks @rhondasu2002 for the detailed bug report! Am digging on this now, though at first glance I wonder if this has anything to do (though I feel it shouldn't) with the fact that you're creating an index="my_index" first, which I don't believe you end up using.

**Comment by nerdai** [comment]
@rhondasu2002: I can confirm I was able to reproduce the bug on my own end. Here are my findings:

- In reviewing the full traceback, I see that `if self.client.indices.exist(index=index_name):` line was returning a `Coroutine` since `client` is `AsyncElasticsearch` and thus needs to be `await`'ed.

I have a new PR that fixes this.

Once the PR lands and the version of llama-index is released, you will need to delete the `paul_graham` index locally before being able to run your script.

```sh
curl -X DELETE "localhost:9200/paul_graham?pretty"
```

If you are using a different port other than 9200, then simply replace it with the one that you are using.
